src/Features/Core/Portable/ImplementAbstractClass/ImplementAbstractClassData.cs

```diff
@@ -0,0 +1,305 @@
+            DeclarationModifiers modifiers, Accessibility accessibility)
+        {
+            var syntaxFacts = _document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var generator = _document.GetRequiredLanguageService<SyntaxGenerator>();
+            var body = throughMember == null
+                ? generator.CreateThrowNotImplementedStatement(compilation)
+                : generator.GenerateDelegateThroughMemberStatement(method, throughMember);
+
+            method = method.EnsureNonConflictingNames(this.ClassType, syntaxFacts);
+
+            return CodeGenerationSymbolFactory.CreateMethodSymbol(
+                method,
+                accessibility: accessibility,
+                modifiers: modifiers,
+                statements: ImmutableArray.Create(body));
+        }
+
+        private IPropertySymbol GenerateProperty(
+            Compilation compilation,
+            IPropertySymbol property,
+            ISymbol? throughMember,
+            DeclarationModifiers modifiers,
+            Accessibility accessibility,
+            ImplementTypePropertyGenerationBehavior propertyGenerationBehavior)
+        {
+            if (property.GetMethod == null)
+            {
+                // Can't generate an auto-prop for a setter-only property.
+                propertyGenerationBehavior = ImplementTypePropertyGenerationBehavior.PreferThrowingProperties;
+            }
+
+            var generator = _document.GetRequiredLanguageService<SyntaxGenerator>();
+            var preferAutoProperties = propertyGenerationBehavior == ImplementTypePropertyGenerationBehavior.PreferAutoProperties;
+
+            var getMethod = ShouldGenerateAccessor(property.GetMethod)
+                ? CodeGenerationSymbolFactory.CreateAccessorSymbol(
+                    property.GetMethod,
+                    attributes: default,
+                    accessibility: property.GetMethod.ComputeResultantAccessibility(this.ClassType),
+                    statements: generator.GetGetAccessorStatements(
+                        compilation, property, throughMember, preferAutoProperties))
+                : null;
+
+            var setMethod = ShouldGenerateAccessor(property.SetMethod)
+                ? CodeGenerationSymbolFactory.CreateAccessorSymbol(
+                    property.SetMethod,
+                    attributes: default,
+                    accessibility: property.SetMethod.ComputeResultantAccessibility(this.ClassType),
+                    statements: generator.GetSetAccessorStatements(
+                        compilation, property, throughMember, preferAutoProperties))
+                : null;
+
+            return CodeGenerationSymbolFactory.CreatePropertySymbol(
+                property,
+                accessibility: accessibility,
+                modifiers: modifiers,
+                getMethod: getMethod,
+                setMethod: setMethod);
```

Comment:
I'd ordinarily remove the argument name if it was equal to or less specific than the expression being passed.